src/com/carolinarollergirls/scoreboard/core/impl/ClockImpl.java

```diff
@@ -150,6 +150,7 @@ public void scoreBoardChange(ScoreBoardEvent event) {
             } else {
                 setMaximumTime(DEFAULT_MAXIMUM_TIME);
             }
+            resetTime();
```

Comment:
This will cause breakage if the ruleset is changed when a clock is running.

And what unexpected time are you seeing? Changing maximum time is supposed to keep timeElapsed constant, which has just been reset to 0 in "Start new game".

Reply:
Rulesets can only be changed at Start New Game.

If you have a 20m period time from a ruleset, the period clock will show as 10m after the State New Game.

Reply:
> Rulesets can only be changed at Start New Game.

That's true right now, but we have already thought about changing that.

>If you have a 20m period time from a ruleset, the period clock will show as 10m after the State New Game.

That's probably a race between capping the time at maximum time and reducing the time on a countdown clock when maximum time is reduced. It seems the former is applied first, while it should be the latter. This should be fixed in that code (especially as it affects more cases).

Reply:
Okay, found that. PTAL.